proto/waste.proto

```diff
@@ -0,0 +1,102 @@
+syntax = "proto3";
+
+package smartcore.bos;
+
+option go_package = "github.com/vanti-dev/sc-bos/pkg/gen";
+
+import "google/protobuf/field_mask.proto";
+import "google/protobuf/timestamp.proto";
+import "types/info.proto";
+import "types/change.proto";
+
+// WasteApi exposes details about units of waste that have been produced by the building
+service WasteApi {
+  rpc ListWasteRecords(ListWasteRecordsRequest) returns (ListWasteRecordsResponse);
+  rpc PullWasteRecords(PullWasteRecordsRequest) returns (stream PullWasteRecordsResponse);
+}
+
+service WasteInfo {
+  rpc DescribeWasteRecord(DescribeWasteRecordRequest) returns (WasteRecordSupport);
+}
+
+enum DisposalMethod {
+  DISPOSAL_UNSPECIFIED = 0;
+  GENERAL_WASTE = 1;
+  MIXED_RECYCLING = 2;
+}
```

Comment:
Enums in protos are complicated. We found that it's better to place them as nested in the primary message for the service - as a child in WasteRecord in this case.

There are all sorts of issues with enums and C++, Java, Go, etc that kind of go away when we do this.